### PR TITLE
Sync ceilometer

### PR DESCRIPTION
**Pull request by aplanas**

### CODE REVIEW

**Comment by toabctl** [comment]
Is there a way to show the diff of the .conf.erb file in github? I just see ```1,079 additions, 870 deletions not shown```

**Comment by aplanas** [comment]
@toabctl No idea, the same is happening in the heat one: https://github.com/crowbar/crowbar-openstack/pull/58

**Comment by vuntz** [comment]
@toabctl @aplanas just look at the patch on your machine, I'd say :/

File: chef/cookbooks/ceilometer/recipes/server.rb [hunk 1]
```diff
@@ -311,8 +311,11 @@
 end
 
 # Cronjob to repair the database and free space for mongodb.  This
-# only makes sense when the time_to_live > 0
-if node[:ceilometer][:use_mongodb] && node[:ceilometer][:database][:time_to_live] > 0
+# only makes sense when the metering_time_to_live or
+# event_time_to_leave > 0
+time_to_live = node[:ceilometer][:database][:metering_time_to_live] > 0 \
```
Comment (jsuchome) [inline chef/cookbooks/ceilometer/recipes/server.rb]: Wouldn't it be better to have the variable name indicate that it's a boolean? (like time_to_live_set or so)

File: chef/data_bags/crowbar/migrate/ceilometer/021_rename_time_to_live_attribute.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    a["database"]["metering_time_to_live"] = a["database"]["time_to_live"]
+    a["database"].delete("time_to_live")
+  end
+  return a, d
+end
+
+def downgrade(ta, td, a, d)
+  if a["database"]["metering_time_to_live"]
+    a["database"]["time_to_live"] = a["database"]["metering_time_to_live"]
+    a["database"].delete("meterin_time_to_live")
```
Comment (jsuchome) [inline chef/data_bags/crowbar/migrate/ceilometer/021_rename_time_to_live_attribute.rb]: typo (``meterin_time_to_live``)

File: chef/data_bags/crowbar/migrate/ceilometer/022_add_event_time_to_live_attribute.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+def upgrade(ta, td, a, d)
+  a["database"]["event_time_to_live"] = ta["database"]["even_time_to_live"]
```
Comment (jsuchome) [inline chef/data_bags/crowbar/migrate/ceilometer/022_add_event_time_to_live_attribute.rb]: typo: ``even_time_to_live``

**Comment by jsuchome** [comment]
Do we need separate migrations for those 2 time to live variables?

**Comment by aplanas** [comment]
@jsuchome One is a rename and another one is new, I feel better separating in two migrations

File: chef/cookbooks/ceilometer/recipes/common.rb [hunk 4]
```diff
@@ -94,7 +100,8 @@
       node_hostname: node["hostname"],
       hypervisor_inspector: hypervisor_inspector,
       libvirt_type: libvirt_type,
-      time_to_live: time_to_live,
+      metering_time_to_live: metering_time_to_live,
+      event_time_to_live: event_time_to_live,
```
Comment (jsuchome) [inline chef/cookbooks/ceilometer/recipes/common.rb]: it looks like in the template you're still using just ``@time_to_live`` value, is that right?

Comment (aplanas) [inline chef/cookbooks/ceilometer/recipes/common.rb]: no, is changed in the same commit: https://github.com/aplanas/crowbar-openstack/commit/9bda334707ea36ecdf3a7b6237e3924bba08511b#diff-8ba6ab0f73ac9ad5fa0f2fe85bb234deR503

Comment (jsuchome) [inline chef/cookbooks/ceilometer/recipes/common.rb]: Ok, sorry, it's confusing without that diff

**Comment by vuntz** [comment]
Can't comment on the right line of the conf file, but this is wrong:
`host_ip = <%= node[:nova][:vcenter][:port] rescue "443" %>`
should be host_port

**Comment by vuntz** [comment]
Shouldn't we set `[api]/workers`?

**Comment by vuntz** [comment]
`time_to_live` is used in the webui, so code needs to be adapted there.

**Comment by vuntz** [comment]
Should we set `shuffle_time_before_polling_task` to something non-0?

**Comment by vuntz** [comment]
I believe pipeline.yaml.erb should also be sync'ed.

**Comment by vuntz** [comment]
There's also a `alarm_history_time_to_live` setting, that we could add with the other ttl settings...

**Comment by vuntz** [comment]
Wondering out loud: do we want to set `refresh_pipeline_cfg`?

**Comment by vuntz** [comment]
Maybe also `batch_polled_samples`.

**Comment by aplanas** [comment]
@vuntz  `alarm_history_time_to_live` is deprecated for removal and is going to be ignored

**Comment by aplanas** [comment]
* Fix typos
* Use `time_to_live_set`
* Fix `host_port` reference
* Set `[api]/workers`
* Export `*_time_to_live` to the WebUI
* Sync `pipeline.yaml.erb`
* Set `shuffle_time_before_polling_task` to 15 seconds (rndint(0, 15))

**Comment by aplanas** [comment]
> Wondering out loud: do we want to `set refresh_pipeline_cfg`?

Maybe we can wait a bit:
https://bugs.launchpad.net/ceilometer/+bug/1477321

**Comment by vuntz** [comment]
> @vuntz alarm_history_time_to_live is deprecated for removal and is going to be ignored

Isn't it deprecated just because of aodh? I guess the question is are we switching to it? :-)

**Comment by vuntz** [comment]
I checked and am not super convinced about the need for changing `batch_polled_samples`. So only question left is the one about `alarm_history_time_to_live` / aodh.

Other than that, LGTM.

**Comment by aplanas** [comment]
+1 Tempest green

**Comment by vuntz** [comment]
+1

**Pull request merged**